api/v1alpha1/dnspolicy_types.go

```diff
@@ -239,33 +234,17 @@ func (l *DNSPolicyList) GetItems() []kuadrant.Policy {
 // By default, this health check will be applied to each unique DNS A Record for
 // the listeners assigned to the target gateway
 type HealthCheckSpec struct {
-	Endpoint                  string           `json:"endpoint,omitempty"`
-	Port                      *int             `json:"port,omitempty"`
-	Protocol                  *string          `json:"protocol,omitempty"`
-	FailureThreshold          *int             `json:"failureThreshold,omitempty"`
-	AdditionalHeadersRef      *string          `json:"additionalHeadersRef,omitempty"`
-	ExpectedResponses         []int            `json:"expectedResponses,omitempty"`
-	AllowInsecureCertificates bool             `json:"allowInsecureCertificates,omitempty"`
-	Interval                  *metav1.Duration `json:"interval,omitempty"`
-}
-
-func (s *HealthCheckSpec) Validate() error {
-	if s.Interval != nil {
-		if s.Interval.Duration < (time.Second * 5) {
-			return fmt.Errorf("invalid value for spec.healthCheckSpec.interval %v, it cannot be shorter than 5s", s.Interval.Duration)
-		}
-	}
-
-	return nil
+	// +kubebuilder:validation:Required
+	Endpoint string `json:"endpoint"`
+	// +kubebuilder:validation:Required
+	Port *int `json:"port"`
+	// +kubebuilder:validation:Required
+	Protocol *string `json:"protocol"`
+	// +kubebuilder:validation:Required
+	FailureThreshold *int `json:"failureThreshold"`
```

Comment:
Should probably avoid using pointers if its required. Also, you don't need these lines `+kubebuilder:validation:Required` see https://github.com/Kuadrant/kuadrant-operator/pull/578